components/org.wso2.carbon.identity.api.server.oauth.scope/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/impl/ScopesApiServiceImpl.java

```diff
@@ -292,23 +290,14 @@ public Response deleteScope(String name) {
      */
     private static URI buildURIForHeader(String scopeName) {
 
-        String tenantQualifiedRelativePath =
-                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
-        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + scopeName, false, true);
-
-        URI location = URI.create(url);
-        if (!location.isAbsolute()) {
-            Message currentMessage = PhaseInterceptorChain.getCurrentMessage();
-            if (currentMessage != null) {
-                UriInfo ui = new UriInfoImpl(currentMessage.getExchange().getInMessage(), null);
-                try {
-                    return new URI(ui.getBaseUri().getScheme(), ui.getBaseUri().getAuthority(), url, null, null);
-                } catch (URISyntaxException e) {
-                    LOG.error("Server encountered an error while building the location URL with scheme: " +
-                            ui.getBaseUri().getScheme() + ", authority: " + ui.getBaseUri().getAuthority() +
-                            ", url: " + url, e);
-                }
-            }
+        URI location;
+        String context = getContext(SERVER_API_PATH_COMPONENT + scopeName);
```

Comment:
better to do it in the method itself than switching to a private method here. It makes the flow easy to understand 

```suggestion
        String context = IdentityTenantUtil.isTenantQualifiedUrlsEnabled() ? endpoint : String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + endpoint;
```